src/System.CommandLine.Tests/Binding/ModelBinderConstructorTests.cs

```diff
@@ -0,0 +1,140 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.CommandLine.Binding;
+using System.CommandLine.Invocation;
+using System.IO;
+using System.Reflection;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Xunit;
+using common = System.CommandLine.Tests.Binding.ModelBindingCommandHandlerTests;
+
+namespace System.CommandLine.Tests.Binding
+{
+    public class ModelBinderConstructorTests
+    {
+
+        [Theory]
+        [InlineData(typeof(ClassWithCtorParameter<int>))]
+        [InlineData(typeof(ClassWithSetter<int>))]
+        [InlineData(typeof(ClassWithCtorParameter<string>))]
+        [InlineData(typeof(ClassWithSetter<string>))]
+        [InlineData(typeof(FileInfo))]
+        [InlineData(typeof(FileInfo[]))]
+        [InlineData(typeof(string[]))]
+        [InlineData(typeof(List<string>))]
+        [InlineData(typeof(int[]))]
+        [InlineData(typeof(List<int>))]
+        public async Task Handler_constructor_receives_option_arguments_bound_to_the_specified_type(
```

Comment:
`TestCommandHandler<T>` has a default constructor so the test name (and presence in the class given the class name) looks inaccurate.

Reply:
I deleted this file because it was identical to tests in ModelBindingCommandHandlerTests. 

We could use more constructor tests, but I think this should correspond to updating the method of binding tests.